examples/with-vitest/vite.config.ts

```diff
@@ -1,33 +1,34 @@
 /// <reference types="vitest" />
-/// <reference types="vite/client" />
 
-import { defineConfig } from "vite";
 import solid from "solid-start";
+import { defineConfig, loadEnv } from "vite";
 
-export default defineConfig({
-  test: {
-    globals: true,
-    environment: "jsdom",
-    transformMode: {
-      web: [/\.[tj]sx?$/]
+export default defineConfig(({ mode }) => {
+	// Load env file based on `mode` in the current working directory.
+	// Set the third parameter to '' to load all env regardless of the `VITE_` prefix.
+	const env = loadEnv(mode, process.cwd(), '');
+
+  return {
+    test: {
+      environment: "jsdom",
+      transformMode: {
+        web: [/\.[tj]sx?$/]
+      },
+      setupFiles: "./vitest.setup.ts",
+      // solid needs to be inline to work around
+      // a resolution issue in vitest:
+      deps: {
+        inline: [/solid-js/]
+      }
```

Comment:
maybe we should do this as part of our vite plugin..

Reply:
Tbh, I'm not completely sure why it is needed in the first place, but without it, the imports from `solid-js/web` aren't working correctly...

Reply:
yeah I think this was figured out by some vitest magician lol.. but its needed.. I think we can do this inside a plugin we include with solid-start.. can you see if putting this whole config in a plugin works that people can install and we will include it by default in solid-start

Reply:
Hmm, don't think it should live inside a plugin for `solid-start` as none of this is specific to `solid-start`. The main culprit is the "import-magic" for ssr in `solid-testing-library` and imho it should be solved there and not hidden here.

It should be possible though…

Reply:
This is because vitest has two mechanisms of loading imports: using vite or using node. Unfortunately, solid's exports lead to them being imported via both over different routes, which breaks reactivity.

Luckily, a PR to fix this is already open.